scripts/manila-share-snapshot.py

```diff
@@ -83,21 +146,95 @@ def get_orphan_snapshots(self):
         with self.orphan_snapshots_lock:
             return list(self.orphan_snapshots.values())
 
+    def sync_share_snapshot_state(self, share_snapshots, dry_run=True):
+        """ Deal with share snapshot in stuck states for more than 15 minutes """
+        msg = "ManilaSyncSnapshotState: snapshot=%s, instance=%s, status=%s"
+        msg_dry_run = "Dry run: " + msg
+        for snapshot in share_snapshots:
+            snapshot_status = snapshot['status']
+            if snapshot_status not in ['creating', 'error_deleting', 'deleting']:
+                continue
+
+            snapshot_id = snapshot['id']
+            instances = self._query_share_snapshot_instances(snapshot_id)
+            if len(instances) == 0:
+                continue
+            elif len(instances) == 1:
+                instance = instances[0]
+                instance_updated_at = instance['updated_at']
+                if instance_updated_at is not None:
+                    if is_utcts_recent(instance_updated_at, 900):
+                        continue
+                else:
+                    continue
+                instance_id = instance['id']
+                instance_status = instance['status']
+                if dry_run:
+                    log.info(msg_dry_run, snapshot_id, instance_id, instance_status)
+                    continue
+                log.info(msg, snapshot_id, instance_id, instance_status)
+                if snapshot_status == 'error_deleting':
+                    self.share_snapshot_force_delete(snapshot_id)
+                    continue
+                self.share_snapshot_reset_state(snapshot_id, 'error')
+                if snapshot_status  == 'deleting':
+                    self.share_snapshot_delete(snapshot_id)
+            else:
+                for instance in instances:
+                    instance_updated_at = instance['updated_at']
+                    if instance_updated_at is not None:
+                        if is_utcts_recent(instance_updated_at, 900):
+                            continue
+                    else:
+                        continue
+                    instance_id = instance['id']
+                    instance_status = instance['status']
+                    if dry_run:
+                        log.info(msg_dry_run, snapshot_id, instance_id, instance_status)
+                        continue
+                    log.info(msg, snapshot_id, instance_id, instance_status)
+                    if snapshot_status == 'error_deleting':
+                        continue
+                    self.share_snapshot_instance_reset_state(instance_id, 'error')
```

Comment:
I have same comments as in the sync_share_instance_state function. And besides, you might have forgot the 'deleting' case after line 198.

Reply:
same as above

Reply:
where is the 'deleting' case then?